File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        pairs = {"{":"}","[":"]","(":")"}
        opening = set(['[','{','('])
        for i in range(len(s)):
            if s[i] in opening:
                stack.append(s[i])
                continue
            elif(stack!=[] and s[i] == pairs[stack[-1]]):
                stack.pop()
            else:
                return False
        return stack==[]    
         
        

File: 20-valid-parentheses/20-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = []
    let hash = {"{":"}","[":"]","(":")"}
    let index=0
    while(index < s.length){
        if("[{(".includes(s[index])){
            stack.push(s[index])
        }
        else if(stack!=[] && hash[stack[stack.length -1]]==s[index]){
            stack.pop()
        }
        else{
            return false
        }
        index+=1
    }
    console.log(stack)
    return stack.length==0
};